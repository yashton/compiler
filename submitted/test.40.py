#
def fact(  x\
):

 
  if x == 1: 
    return 1    #this is comment to ignore

#  elif x ==0:    

#    return 1
  else:

        return x* fact((x+(3
        
-1)+3))#this is comment to ignore