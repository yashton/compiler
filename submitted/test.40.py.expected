(KEYWORD def)
(ID "fact")
(PUNCT "(")
(ID "x")
(PUNCT ")")
(PUNCT ":")
(NEWLINE)
(INDENT)
(KEYWORD if)
(ID "x")
(PUNCT "==")
(LIT 1)
(PUNCT ":")
(NEWLINE)
(INDENT)
(KEYWORD return)
(LIT 1)
(NEWLINE)
(DEDENT)
(KEYWORD else)
(PUNCT ":")
(NEWLINE)
(INDENT)
(KEYWORD return)
(ID "x")
(PUNCT "*")
(ID "fact")
(PUNCT "(")
(PUNCT "(")
(ID "x")
(PUNCT "+")
(PUNCT "(")
(LIT 3)
(PUNCT "-")
(LIT 1)
(PUNCT ")")
(PUNCT "+")
(LIT 3)
(PUNCT ")")
(PUNCT ")")
(DEDENT)
(DEDENT)
(ENDMARKER)
